psycop/common/model_training_v2/loggers/mlflow_logger.py

```diff
@@ -20,20 +21,26 @@ def sanitise_dict_keys(d: dict[str, Any]) -> dict[str, Any]:
 
 
 @BaselineRegistry.loggers.register("mlflow_logger")
+@dataclass
 class MLFlowLogger(BaselineLogger):
-    def __init__(
-        self, experiment_name: str, tracking_uri: str = "http://exrhel0371.it.rm.dk:5050"
-    ) -> None:
-        mlflow.set_tracking_uri(tracking_uri)
-
-        # Start a new run. End a run if it already exists within the process.
-        if mlflow.active_run() is not None:
-            mlflow.end_run()
-
-        self.mlflow_experiment = mlflow.set_experiment(experiment_name=experiment_name)
-        self.experiment_id = self.mlflow_experiment.experiment_id
-
-        self._log_str = ""
+    experiment_name: str
+    tracking_uri: str = "http://exrhel0371.it.rm.dk:5050"
+    postpone_run_creation_to_first_log: bool = False
+    _run_initialised: bool = False
```

Comment:
Having a private argname in the init feels very strange to me - I guess this is one case where dataclasses aren't great. If it wasn't a dataclass we could just do `_run_initalised = False` in init which would be nicer imo